test/unit/verifier/lint/test_precommit.py

```diff
@@ -0,0 +1,181 @@
+#  Copyright (c) 2015-2018 Cisco Systems, Inc.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to
+#  deal in the Software without restriction, including without limitation the
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
+import pytest
+import sh
+
+from molecule import config
+from molecule.verifier.lint import precommit
+
+
+@pytest.fixture
+def _patched_get_tests(mocker):
```

Comment:
could you please use buitin fixture for this?
```suggestion
def _patched_get_tests(mokeypatch):
```

Reply:
I'm assuming you mean `monkeypatch` here? I am going to need more of an example here to make the conversion there's only two references to `monkeypatch` in the unit tests and the are manipulating environment variables not methods in a class. I'm pretty used to dealing with `mock` and `patch` to fake class methods seems to work okay for me.

Reply:
mock has certain bugs with unpleasant side-effects + there's not much point in having multiple tools doing the same. Why use extra dependency if there's a built-in thing which is officially recommend to use by pytest devs? That's why I'm trying to encourage folks to use `monkeypatch` now.

Reply:
https://docs.pytest.org/en/latest/monkeypatch.html

Reply:
Just FYI `mock` is now part of core https://docs.python.org/3/library/unittest.mock.html so there's no extra dependency anymore.

Could you elaborate on the unpleasant side-effects?

Reply:
You use `pytest-mock`, which provides `mocker` fixture. It's not exactly stdlib mock.

Side-effects is that it breaks pytest itself sometimes. Longer explanation: https://github.com/ansible/molecule/pull/1533#discussion_r260881603